### PR TITLE
fix: isolate missing quorums in InstantSend batches

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
A peer can send a structurally valid ISDLOCK with the genesis block as its cycle hash and an arbitrary parseable BLS signature. The receive path accepts the known cycle-boundary hash, but a mature mainnet node selects signing height 287, before DIP0024 activation, so no quorum is available. The current early return then discards the entire dequeued batch, including unrelated locks that could otherwise be processed.

## What was done?
Mark only the affected lock as a failed message and continue building the batch. Existing verification-result handling keeps that lock unaccepted and routes it through the previous-active-set retry without penalizing its peer solely for the unavailable quorum.

Add a receive-path regression that serializes a peer-supplied genesis-cycle lock, passes it through `NetInstantSend::ProcessMessage`, fetches the real pending queue, and verifies missing-quorum handling without peer punishment. This uses a mined regtest chain and does not inject quorum or recovered-signature state.

Retain the batch-isolation regression placing an unavailable-quorum lock between two locks with cached recovered signatures, including one from the same peer. Cover rotating and non-rotating quorum selection, the retry path, and the absence of acceptance or reconstructed signatures for the failed lock.

## How Has This Been Tested?
- Local macOS arm64 build with prebuilt depends: `make -j8`.
- `./src/test/test_dash --run_test=evo_islock_tests --log_level=message`.
- All 10 cases in `evo_islock_tests` pass. The receive-path regression fails at both expected failure-set checks when the original `return nullptr` is restored, and passes with the fix.
- Changed-line clang-format, whitespace/assertion lint, and independent security/code review.

## Breaking Changes
None. Signature verification and quorum selection rules are unchanged.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests

This pull request was created by Codex.

### CODE REVIEW

File: src/instantsend/net_instantsend.cpp [hunk 1]
```diff
@@ -152,8 +152,9 @@ std::unique_ptr<NetInstantSend::BatchVerificationData> NetInstantSend::BuildVeri
                                               : m_qman.GetQuorum(llmq_params.type, islock->cycleHash);
 
         if (!quorum) {
-            // should not happen, but if one fails to select, all others will also fail to select
-            return nullptr;
+            // Different cycle hashes can select different quorums within the same batch.
+            data->batchVerifier.badMessages.emplace(hash);
+            continue;
```
Comment (knst) [inline src/instantsend/net_instantsend.cpp:157]: the regression test in evo_islock_tests.cpp doesn't proove that it can _indeed_ happen because it creates artificial batch.
This `return nullptr` is the safe analogue of `assert(false);`, not a corner case that happens on production if that's not proven otherwise.

if that's not theoretical case when it can indeed happen, I'd like to see a scenario.

**Review by knst — CHANGES_REQUESTED** [review]
see comment, I believe it's a fix for hypotetical situation when code around is broken, not a normal scenarios.

File: src/instantsend/net_instantsend.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/instantsend/net_instantsend.cpp:157]: The original test did not establish receive-path reachability; that is a valid gap. There is a concrete peer-supplied scenario that does not require broken quorum state:

1. On a synced mainnet node with InstantSend enabled, a peer sends an ISDLOCK with a nonzero txid, one unique input, the genesis block as `cycleHash`, and a parseable BLS signature made with its own key. No valid quorum signature is needed to enter the pending queue.
2. `ProcessMessage` checks the lock structure, resolves genesis to height 0, and accepts `0 % 288 == 0`. It does not require the supplied cycle to be after DIP0024 activation or to have a quorum.
3. At the current tip, `BuildVerificationBatch` computes `nSignHeight = 0 + 288 - 1 = 287`. `ScanQuorums` returns empty because `LLMQ_60_75` was not enabled at that height (mainnet DIP0024 activates at 1,737,792), so quorum selection returns null. Retrying with offset 288 also returns null because the resulting start height is -1.
4. A legitimate lock from a recent cycle in the same dequeued batch selects a different signing height and can have an available quorum. With the original early return, none of the batch reaches `ApplyVerificationResults`, and the empty failure set also skips retry.

The added `received_genesis_cycle_has_no_quorum` test serializes this kind of input through `NetInstantSend::ProcessMessage` on a mined regtest chain, retrieves it from the real pending queue, and checks missing-quorum handling on both attempts with no peer punishment. It does not inject quorum or recovered-signature state. The existing mixed-batch test separately checks isolation of unaffected entries; this is not a full multi-node reproduction.

Validation: full local macOS arm64 `make -j8`; all 10 `evo_islock_tests` cases pass. Restoring the original `return nullptr` makes the new receive-path test fail at both expected failure-set checks. Changed-line clang-format, whitespace and assertion lint pass.

I kept the production fix: this is reachable with untrusted network input, so an invariant failure elsewhere is not a prerequisite.

---
🤖 Posted autonomously by Codex on behalf of pasta.

**Review by thepastaclaw — APPROVED** [review]
## Final validation — Phase 2 only (queue backlog)

Verified the complete two-commit range at head 71cf6252599d1e716e0f5e68628eb9cddd935069; no in-scope defects were found. The fix preserves per-lock failure state through batch verification and retry handling without penalizing peers solely for unavailable quorums, and the complementary regressions cover receive-path reachability and mixed-batch isolation across rotating and non-rotating configurations. Commit scope matches the stated purpose, and git diff --check passed; reviewer-reported build and test results were not independently rerun.

Source: reviewer 1: `gpt-6-astra` (agent: `phase2-reviewer`, role: `general`); reviewer 2: `gpt-6-astra` (agent: `phase2-reviewer`, role: `dash-core-commit-history`); final verifier: `gpt-6-astra` (agent: `astra-verifier`, role: `final-verifier`)

### Review provenance
- Triage: `critical` by `gpt-6-astra` (effort low) — Although the production diff is small, it changes peer-controlled InstantSend batch processing and signature-verification failure handling, where incorrect isolation, retries, or peer penalties could disrupt transaction locking or enable denial of service.
- Phase 1 reviewers: **not run (skipped for throughput: 25 PRs queued, above the 10 limit)**
- Fresh verifier: `gpt-6-astra` — final-verifier; agent `astra-verifier`
- Phase 2 reviewers: `gpt-6-astra` — general (completed, effort xhigh); agent `phase2-reviewer`, `gpt-6-astra` — dash-core-commit-history (completed, effort xhigh); agent `phase2-reviewer`
